pkgs/development/python-modules/devito/default.nix

```
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, anytree
+, nbval
+, flake8
+, sympy
+, scipy
+, cached-property
+, psutil
+, py-cpuinfo
+, cgen
+, click
+, multidict
+, distributed
+, pyrevolve
+, codepy
+, pytest
+, pytest-xdist
+, matplotlib
+, mpi
+}:
+
+buildPythonPackage rec {
+  pname = "devito";
+  version = "4.6.2";
+
+  src = fetchFromGitHub {
+    owner = "devitocodes";
+    repo = "devito";
+    rev = "v${version}";
+    sha256 = "sha256-1yPKaVtxjeoANCUAqJYZndi9C/1BVVg5vuNNAb19zvs=";
+  };
+
+  patches = [ ./clear-deps.patch ];
+
+  postPatch = ''
+    substituteInPlace devito/arch/compiler.py --replace \
+    "obj = super().__new__(cls, *args, **kwargs)" \
+    "obj = super().__new__(cls)"
```

```suggestion
    substituteInPlace devito/arch/compiler.py \
      --replace "obj = super().__new__(cls, *args, **kwargs)" "obj = super().__new__(cls)"
```
Please add a short comment why we need this.